src/com/talar/server/ServerProcess.java

```diff
@@ -0,0 +1,250 @@
+package com.talar.server;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+public class ServerProcess extends Thread {
+
+        private Socket socket = null;
+        private BufferedInputStream bis = null;
+        private BufferedOutputStream bos = null;
+        private boolean clientOn = false;
+        private boolean serverOn = false;
+
+        public ServerProcess(Socket socket) {
+                this.socket = socket;
+                clientOn = true;
+                serverOn = true;
+                
+                InetAddress addr = socket.getInetAddress();
+                String ipAddr = addr.getHostAddress();
+                System.out.println("Client connected " + ipAddr + ":" + socket.getLocalPort());
+        }
+        
+        public void init() throws IOException {
+                InputStream in = socket.getInputStream();
+                OutputStream out = socket.getOutputStream();
+                bis = new BufferedInputStream(in);
+                bos = new BufferedOutputStream(out);
+        }
+
+        @Override
+        public void run() {
+                while(isClientOn()) {
+                        String message = readInput();
+                        System.out.println("Server Received: " + message);
+
+                        String response = getResponse(message);
+                        System.out.println("Response: " + response);
+                        writeOutput(response);
+                }
+
+                close();
+        }
+
+        private String readInput() {
+                String message = "";
+
+                try {
+                        StringBuilder sb = new StringBuilder();
+                        
+                        int c = 0;
+                        while(true) {
+                                
+                                c = bis.read();
+
+                                if(c == 0) {
+                                        break;
+                                }
+
+                                sb.append((char)c);
+                        }
+
+                        message = sb.toString();
+                }
+                catch(IOException e) {
+                        e.printStackTrace();
```

Comment:
You might want to consider what happens when an exception is caught. If this fails half way then the StringBuilder would have half the message inside it and the method will return corrupted data. You should re-initialize your return value to something that indicates an exception occurred like a null value. Or... you could propagate the exception upwards and handle it there.